tests/s3.bats

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bats
+
+load helpers.sh
+
+# Upload
+
+@test "uploading blob to test bucket" {
+  run UPLOAD "$CACHE/upload?bucket=test" -F "files=@$DIR/blob"
+  [[ "$status" -eq 0 ]]
+  [[ "$output" == "200" ]]
+
+  run AWS s3 ls s3://test/blob
+  [[ "$status" -eq 0 ]]
+}
+
+@test "uploading blob to test bucket with paths" {
+  run UPLOAD "$CACHE/upload?bucket=test&path=folder" -F "files=@$DIR/blob"
+  [[ "$status" -eq 0 ]]
+  [[ "$output" == "200" ]]
+
+  run AWS s3 ls s3://test/folder/blob
+  [[ "$status" -eq 0 ]]
+
+  run UPLOAD "$CACHE/upload?bucket=test&path=folder/subfolder" -F "files=@$DIR/blob"
+  [[ "$status" -eq 0 ]]
+  [[ "$output" == "200" ]]
+
+  run AWS s3 ls s3://test/folder/subfolder/blob
+  [[ "$status" -eq 0 ]]
+}
+
+# Delete
+
+@test "deleting blob from test bucket" {
+  run DELETE "$CACHE/delete?bucket=test&key=folder/subfolder/blob"
+  [[ "$status" -eq 0 ]]
+  [[ "$output" == "200" ]]
+
+  run AWS s3 ls s3://test/folder/subfolder/blob
+  [ "$status" -eq 1 ]
+}
+
+@test "deleting blob prefix from test bucket" {
+  run DELETE "$CACHE/delete?bucket=test&prefix=folder/subfolder"
```

Comment:
Move bucket name to var across all these functions ?

Reply:
Done